tests/RobotFramework/tedge/call_tedge_config_list.robot

```diff
@@ -0,0 +1,148 @@
+*** Settings ***
+Documentation    Purpose of this test is to verify that the tedge config list and tedge config list --all
+...              will result with Return Code 0
+...              Set new device type and return to default value
+...              Set new kay path and return to default value
+...              Set new cert path and return to default value
+...              Set new c8y.root.cert.path and return to default value
+...              Set new c8y.smartrest.templates and return to default value
+...              Set new az.root.cert.path and return to default value
+...              Set new az.mapper.timestamp and return to default value
+...              Set new mqtt.bind_address and return to default value
+...              Set new mqtt.port and return to default value
+...              Set new tmp.path and return to default value
+...              Set new logs.path and return to default value
+...              Set new run.path and return to default value
+
+Library    SSHLibrary
+
+Suite Setup            Open Connection And Log In
+Suite Teardown         SSHLibrary.Close All Connections
+
+*** Variables ***
+${HOST}           192.168.99.110    #Insert the IP address if the default should not be used
+${USERNAME}       pi    #Insert the username if the default should not be used
+${PASSWORD}       thinedge    #Insert the password if the default should not be used
+
+*** Tasks ***
```

Comment:
I feel also add a test for `device.id`, this is not settable. So, one can test this feature.
For example

`sudo tedge config set device.id test`

Error: failed to set the configuration key: device.id with value: test.

Caused by:
    Readonly setting: The device id is read from the device certificate and cannot be set directly.
    To set 'device.id' to some <id>, you can use `tedge cert create --device-id <id>`.